AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m

```diff
@@ -194,13 +232,69 @@ - (BOOL)setupAudioEngine {
      * audio data from the Video SDK and mix it in MainMixerNode. Here we connect the input node to the main mixer node.
      * InputNode -> MainMixer -> OutputNode
      */
-    [_engine connect:_engine.inputNode to:_engine.mainMixerNode format:format];
+    [_recordEngine connect:_recordEngine.inputNode to:_recordEngine.mainMixerNode format:format];
 
-    _renderingContext->renderBlock = (__bridge void *)(_engine.manualRenderingBlock);
+    _capturingContext->renderBlock = (__bridge void *)(_recordEngine.manualRenderingBlock);
```

Comment:
It looks like the context's block is also set here. This would be the correct place IMHO.